src/com/amazon/ion/benchmark/IonSchemaUtilities.java

```diff
@@ -101,25 +111,73 @@ public static int parseConstraints(IonStruct value, String keyWord) throws IOExc
+     * Parse the field 'annotations' based on the provided constraints (required|ordered).
+     * @param constraintStruct contains the top-level constraints of Ion Schema.
+     * @return IonList which contains annotations
+     * @throws IOException if error occur when reading constraints.
      */
-    public static Map<String, Object> getAnnotation(IonStruct value) {
-        Map<String, Object> annotationMap = new HashMap<>();
-        IonValue annotations = value.get(KEYWORD_ANNOTATIONS);
-        if (annotations != null) {
-            IonList annotationList = (IonList) value.get(KEYWORD_ANNOTATIONS);
-            if (annotations.getTypeAnnotations().length != 0) {
-                String annotationConstraint = annotations.getTypeAnnotations()[0];
-                annotationMap.put(KEYWORD_CONSTRAINT, annotationConstraint);
+    public static IonList getAnnotation(IonStruct constraintStruct) throws IOException {
+        IonList annotationList = null;
+        Random random = new Random();
+        try (IonReader annotationInfo = IonReaderBuilder.standard().build(constraintStruct)){
+            annotationInfo.next();
+            annotationInfo.stepIn();
+            while (annotationInfo.next() != null) {
+                if (annotationInfo.getFieldName().equals(IonSchemaUtilities.KEYWORD_ANNOTATIONS)) {
+                    IonValue annotationValue = ReadGeneralConstraints.SYSTEM.newValue(annotationInfo);
+                    List<String> constraint = Arrays.asList(annotationValue.getTypeAnnotations());
+                    IonList annotations = (IonList) annotationValue;
+                    annotationList = checkOrdered(constraint, annotations, random);
+                }
+            }
+        }
+        return annotationList;
+    }
+
+    /**
+     * This is a helper method of getAnnotation which processes the constraint 'Ordered'.
+     * @param constraint is a List which contains all annotations of 'annotations' field.
+     * @param annotationList is the original annotation List without any consideration about constraints.
+     * @param random is a random integer generator.
+     * @return a List of processed annotations.
+     * @throws IOException if error occur when reading constraints.
+     */
+    private static IonList checkOrdered(List<String> constraint, IonList annotationList, Random random) throws IOException {
+        IonList result = annotationList;
+        if (!constraint.contains(IonSchemaUtilities.KEYWORD_ORDERED)) {
+            List<IonValue> annotations = annotationList.stream().collect(Collectors.toList());
+            Collections.shuffle(annotations);
+            try (IonReader shuffledAnnotationReader = IonReaderBuilder.standard().build(annotations.toString())) {
+                shuffledAnnotationReader.next();
+                result = (IonList) ReadGeneralConstraints.SYSTEM.newValue(shuffledAnnotationReader);
+            }
```

Comment:
Rather than writing and re-reading the list to get an IonList, just create an IonList and add the elements of subAnnotationList to it. See IonSystem.newEmptyList.